update.sh

```diff
@@ -49,15 +49,47 @@ echo "// WARNING! generated file do not edit" > src/apis/standard/mod.rs
 for API in "${STANDARD_APIS[@]}"
 do
     echo "generating standard api ${API}"
-    curl -sSL "https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/crd/standard/gateway.networking.k8s.io_${API}.yaml?ref=${VERSION}" | kopium -Af - > src/apis/standard/${API}.rs
+    curl -sSL "https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/crd/standard/gateway.networking.k8s.io_${API}.yaml?ref=${VERSION}" | kopium --schema=derived --derive=JsonSchema --derive=Default --docs -f - > src/apis/standard/${API}.rs
     echo "pub mod ${API};" >> src/apis/standard/mod.rs
 done
 
+ENUMS=(
+    HTTPRouteRulesFiltersRequestRedirectPathType=ReplaceFullPath
+    HTTPRouteRulesFiltersUrlRewritePathType=ReplaceFullPath
+    HTTPRouteRulesFiltersType=RequestHeaderModifier
+    HTTPRouteRulesBackendRefsFiltersRequestRedirectPathType=ReplaceFullPath
+    HTTPRouteRulesBackendRefsFiltersUrlRewritePathType=ReplaceFullPath
+    HTTPRouteRulesBackendRefsFiltersType=RequestHeaderModifier
+)
+
+ENUMS_WITH_DEFAULTS=$(printf ",%s" "${ENUMS[@]}")
+ENUMS_WITH_DEFAULTS=${ENUMS_WITH_DEFAULTS:1}
+GATEWAY_API_ENUMS=${ENUMS_WITH_DEFAULTS} cargo run --manifest-path ./tools/enum_default_generator/Cargo.toml >> src/apis/standard/enum_defaults.rs
+echo "mod enum_defaults;" >> src/apis/standard/mod.rs
+
 echo "// WARNING! generated file do not edit" > src/apis/experimental/mod.rs
 
 for API in "${EXPERIMENTAL_APIS[@]}"
 do
     echo "generating experimental api $API"
-    curl -sSL "https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/crd/experimental/gateway.networking.k8s.io_${API}.yaml?ref=${VERSION}" | kopium -Af - > src/apis/experimental/${API}.rs
+    curl -sSL "https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/crd/experimental/gateway.networking.k8s.io_${API}.yaml?ref=${VERSION}" | kopium --schema=derived --derive=JsonSchema --derive=Default --docs -f - > src/apis/experimental/${API}.rs
     echo "pub mod ${API};" >> src/apis/experimental/mod.rs
 done
+
+ENUMS=(
+    HTTPRouteRulesFiltersRequestRedirectPathType=ReplaceFullPath
+    HTTPRouteRulesFiltersUrlRewritePathType=ReplaceFullPath
+    HTTPRouteRulesFiltersType=RequestHeaderModifier
+    HTTPRouteRulesBackendRefsFiltersRequestRedirectPathType=ReplaceFullPath
+    HTTPRouteRulesBackendRefsFiltersUrlRewritePathType=ReplaceFullPath
+    HTTPRouteRulesBackendRefsFiltersType=RequestHeaderModifier
+    GRPCRouteRulesFiltersType=RequestHeaderModifier
+    GRPCRouteRulesBackendRefsFiltersType=RequestHeaderModifier
+)
+
+ENUMS_WITH_DEFAULTS=$(printf ",%s" "${ENUMS[@]}")
+ENUMS_WITH_DEFAULTS=${ENUMS_WITH_DEFAULTS:1}
+GATEWAY_API_ENUMS=${ENUMS_WITH_DEFAULTS} cargo run --manifest-path ./tools/enum_default_generator/Cargo.toml >> src/apis/experimental/enum_defaults.rs
+echo "mod enum_defaults;" >> src/apis/experimental/mod.rs
+
+cargo fmt
```

Comment:
This is kinda the "main starting point" for new contributors (which in time, I hope to change (e.g. our build be in an xtask instead), so let's make sure we're adding lots of verbose comments to all this stuff.